modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/sql/ClientSqlExecuteBatchRequest.java

```diff
@@ -102,16 +103,18 @@ public static CompletableFuture<Void> process(
     private static void writeBatchResult(
             ClientMessagePacker out,
             long[] affectedRows,
-            Short errorCode,
-            String errorMessage) {
+            int errorCode,
+            String errorMessage,
+            UUID traceId) {
         out.packNil(); // resourceId
 
         out.packBoolean(false); // has row set
         out.packBoolean(false); // has more pages
         out.packBoolean(false); // was applied
         out.packLongArray(affectedRows); // affected rows
-        out.packShort(errorCode); // error code
+        out.packInt(errorCode); // error code
```

Comment:
We have standardized error serialization in https://github.com/apache/ignite-3/blob/d3ab33124127dd18f36727148100d632fffbd02e/modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientInboundMessageHandler.java#L498 .

Let's remove error details from this response, and return failed future instead.

Reply:
For SqlBatchException this doesn't work very well right now.
This exception includes an "update counter" which is lost during serialization.
This counter can be used by the user to understand how many updates were completed successfully (we have several tests for this).

Reply:
Ok, let's handle this as a separate ticket: https://issues.apache.org/jira/browse/IGNITE-22569